book/src/spec.md

```diff
@@ -7,9 +7,9 @@
 We use Halo2/IPA with [Pasta curves](https://github.com/zcash/pasta) developed by Zcash to instantiate our proving system.
 
 ### 1.1 Circuits
-Let `C(x; w) ⟶ 0/1` be a circuit with up to `m` gates. The circuit is represented as polynomials over the chosen curve's **scalar field**, following [plonk-ish arithmetization](https://zcash.github.io/halo2/concepts/arithmetization.html). Commitments are generated over the curve's **base field**.
+Let `C(x; w) ⟶ 0/1` be a circuit. An elliptic curve has arithmetic defined within the _scalar field_ (i.e. the prime number of points of the curve). When we formulate circuits, we use this scalar field. The circuit is represented as polynomials over the chosen curve's scalar field, following [plonk-ish arithmetization](https://zcash.github.io/halo2/concepts/arithmetization.html). When we want to _commit_ to these values in the scalar field, we end up with values in the _base field_. When using these committed values, we encounter what is known as _non-native arithmetic_. This is the motivating factor for proposing a _cycle of curves_.
```

Comment:
What does "has arithmetic defined within scalar field" mean? I think arithmetic is usually defined (although I'm not sure what does it mean to define arithmetic) through manipulating with coordinates which belong to the base field, so this definition is a bit confusing. I'm generally not sure if we need to explain what an elliptic curve's scalar field is here. Maybe we could just give a link to scalar field definition?

Also I think non-native field arithmetic is the motivating factor for using more than one curve, not necessarily cycle of curves. Cycles are helpful when we have more than 1-layer recursion (our  current case) or just generally switch between fields too often, otherwise we could have used the three curves we used in the zk-garage version and be fine

Reply:
> What does "has arithmetic defined within scalar field" mean?

For your first question, it means that in kP where k is a scalar and P is a point, k is a scalar field. I took that exact phrase from this post, which I think it was very nicely and succinctly summarised https://blog.icme.io/can-we-avoid-cycles-of-curves/

>  I'm generally not sure if we need to explain what an elliptic curve's scalar field is here. 

The specs used to say "Commitments are generated over the curve's **base field**" which doesn't say a commitment of what. I just added a few clarifications and motivation for our specific example. I don't think adding "(i.e. the prime number of points of the curve)" is not a problem.

> Also I think non-native field arithmetic is the motivating factor for using more than one curve

Sure, it is one of the solutions to the problem. But it is _exactly_ the solution that Halo2 takes, which is what we choose here. If we ever move away from Halo2 we can change it (together with most of the specs). I don't see the problem now

Reply:
> For your first question, I wrote "has arithmetic defined within the scalar field". I could have said the arithmetic of the elliptic curve as a group but I found it redundant. I can add it though if it helps.

Yeah I agree it is redundant, I just don't fully understand this way of phrasing

>I don't think adding "(i.e. the prime number of points of the curve)" is a problem.

No, of course it isn't!

> I don't see the problem now

I find it a bit confusing. Maybe we could just elaborate further, e.g. " to solve the non-native field arithmetic problem, we can use another curve whose scalar field is the base field of the main curve. Cycles of curves allow to switch between fields multiple times without the need to introduce new curve each time"